src/boot/bootctl.c

```diff
@@ -309,6 +309,34 @@ static int status_variables(void) {
         return 0;
 }
 
+static int boot_entry_file_check(const char *root, const char *p) {
+        _cleanup_free_ char *path;
+
+        path = path_join(root, p);
+        if (!path)
+                return log_oom();
+
+        if (access(path, F_OK) < 0)
+                return -errno;
+
+        return 0;
+}
+
+static void boot_entry_file_list(const char *field, const char *root, const char *p) {
+        int status = boot_entry_file_check(root, p);
+        const char
+                *on =    status == 0 ? "" : ansi_highlight_red(),
+                *off =   status == 0 ? "" : ansi_normal(),
+                *open =  status == 0 ? "" : " (",
+                *msg =   status == 0 ? "" : strerror(-status),
+                *close = status == 0 ? "" : ")";
+
```

Comment:
i think it would be nicer to write this like this:

```c
int status;

print("%13s%s", strempty(field), field ? ":" : " ");
staus = boot_entry_file_check(root, p);
if (status < 0) {
        errno = -status;
        printf("%s%m%s (%m)\n", ansi_highlight_red(), p, ansi_normal());
} else
        printf("%s\n", p);
```

That uses a single comparison and I think is a lot more readable..